### ISSUE TITLE
Building on Debian 12 Bookworm needs autoreconf step

### ISSUE THREAD

**Issue by damienvancouver** [nethappen/blocksync-fast#5]
When compiling on Debian 12 bookworm (stable), running "./configure" then "make"  yields the error:

WARNING: 'automake-1.17' is missing on your system.
         You should only need it if you modified 'Makefile.am' or
         'configure.ac' or m4 files included by 'configure.ac'.
make: *** [Makefile:340: Makefile.in] Error 1

Running:  `autoreconf -f -i` fixes it, then `make` correctly builds.   Debian bookworm has automake 1.16.5-1.3    so that is probably the reason.

we could just mention this extra step in the build instructions as one way around this!

**Comment by nethappen** [comment]
Yes, I just got feedback recently that it stopped compiling e.g. on the latest Archlinux version.
So I did `autoreconf --install` on the latest Archlinux and tested it on Gentoo too and included this working new version.
I see that it doesn't work on older Debians, maybe on others somewhere else too. `autoreconf -f -i` fixes the problem.
But we need to find a universal solution to this problem.
